packages/playwright-core/src/utils/wsServer.ts

```diff
@@ -77,7 +77,7 @@ export class WSServer {
           reject(new Error('Could not bind server socket'));
           return;
         }
-        const wsEndpoint = typeof address === 'string' ? `${address}${path}` : `ws://${hostname || 'localhost'}:${address.port}${path}`;
+        const wsEndpoint = typeof address === 'string' ? `${address}${path}` : `ws://${hostname || address.address}:${address.port}${path}`;
```

Comment:
For compatibility, I only changed the `host` option document to unspecified and specified the output ws hostname to be the address output by the server.